app/services/case_validate_rejected_offender_sar_service.rb

```diff
@@ -32,6 +32,6 @@ def call(message = nil)
 private
 
   def update_case_number
-    @case.number.tr!("R", "")
+    @case.number.tr!(@case.number, @case.set_valid_case_number)
```

Comment:
You should be able to just call `@case.set_valid_case_number`. 

Also, as the method name is clear, it can probably be called directly from the call method.